source/connect/connection-targets.txt

```diff
@@ -0,0 +1,139 @@
+  connect to a MongoDB server running on ``localhost`` on port ``27017``:
+
+  .. literalinclude:: /includes/connect/connection-targets-local-replica.scala
+   :language: scala
+   :start-after: start-connect-local
+   :end-before: end-connect-local
+   :dedent:
+
+- Explicitly specify the ``hostname`` to connect to a MongoDB
+  instance running on the specified host on port ``27017``: 
+
+  .. literalinclude:: /includes/connect/connection-targets-local-replica.scala
+   :language: scala
+   :start-after: start-connect-local-host
+   :end-before: end-connect-local-host
+   :dedent:  
+
+- Explicitly specify the ``hostname`` and the ``port``:
+
+  .. literalinclude:: /includes/connect/connection-targets-local-replica.scala
+   :language: scala
+   :start-after: start-connect-local-host-port
+   :end-before: end-connect-local-host-port
+   :dedent:
+
+Replica Sets
+------------
+
+To connect to a replica set, specify the hostnames (or IP addresses) and 
+port numbers of the replica set members.
+
+If you aren't able to provide a full list of hosts in the replica set, you can 
+specify one or more of the hosts in the replica set and instruct the {+driver-short+} to 
+perform automatic discovery to find the others. To instruct the driver to perform
+automatic discovery, perform one of the following actions:
+
+- Specify the name of the replica set as the value of the ``replicaSet`` parameter.
+- Specify ``false`` as the value of the ``directConnection`` parameter.
+- Specify more than one host in the replica set.
+
+You can connect to a MongoDB replica set by specifying the members in a 
+``ConnectionString``. The following example shows how to specify three members 
+of the replica set:
+
+.. literalinclude:: /includes/connect/connection-targets-local-replica.scala
+   :language: scala
+   :start-after: start-replica-set
+   :end-before: end-replica-set
+   :dedent:
+
+The following example shows how to specify members of the replica set and the 
+``replicaSet`` option with the replica set name:
+
+.. literalinclude:: /includes/connect/connection-targets-local-replica.scala
+   :language: scala
+   :start-after: start-replica-set-option
+   :end-before: end-replica-set-option
+   :dedent:
+
+The following example shows how to specify a list of ``ServerAddress`` instances 
+corresponding to all of the replica set members:
+
+.. literalinclude:: /includes/connect/connection-targets-local-replica.scala
+   :language: scala
+   :start-after: start-replica-set-server-address
+   :end-before: end-replica-set-server-address
+   :dedent:
+
+.. note::
+
+   The ``MongoClient`` constructor is *non-blocking*.
+   When you connect to a replica set, the constructor returns immediately while the
+   client uses background threads to connect to the replica set.
+   
+   If you construct a ``MongoClient`` and immediately print the string representation
+   of its ``nodes`` attribute, the list might be empty while the client connects to
+   the replica-set members.
```

Comment:
[s] Can potentially link to any relevant API documentation here? It looks like other drivers are linking to `MongoClient` documentation but not sure if that applies for Scala or not 😅